linkerd/app/inbound/src/http/router.rs

```diff
@@ -417,9 +418,60 @@ impl tap::Inspect for Logical {
     }
 
     fn route_labels<B>(&self, req: &http::Request<B>) -> Option<tap::Labels> {
-        req.extensions()
+        let mut labels = std::collections::BTreeMap::new();
+        for (k, v) in req
+            .extensions()
             .get::<profiles::http::Route>()
-            .map(|r| r.labels().clone())
+            .iter()
+            .flat_map(|route| route.labels().iter())
+        {
+            labels.insert(k.clone(), v.clone());
+        }
+        if let Some(route_labels) = req.extensions().get::<RouteLabels>() {
+            labels.insert(
```

Comment:
nit/TIOLI: since we're always adding 6 labels in this branch, we could write
```suggestion
            labels.reserve(6);
            labels.insert(
```
so that we only reallocate once.

alternatively, using `labels.extend(&[...])` instead would also do this.